plugins/antiflood.ts

```diff
@@ -0,0 +1,35 @@
+import { createPlugin } from "../core/plugins.ts";
+import { Queue } from "../deps.ts";
+
+interface AntiFloodFeatures {
+  options: {
+    /** Milliseconds to wait between dispatching private messages.
+     *
+     * Defaults to 0 milliseconds (no delay) */
+    floodDelay?: number;
+  };
+}
+
+export default createPlugin("antiflood", [])<AntiFloodFeatures>(
+  (client, options) => {
+    // Queue object and delay structure for anti-flood protection
+    const queue = new Queue();
+    const delay = (ms = 0) => new Promise((resolve) => setTimeout(resolve, ms));
+
+    // Queues up limiter for outgoing messages with before and after send hooks
+
+    client.hooks.hookCall("send", async (send, command, ...params) => {
+      if (command === "PRIVMSG" && options.floodDelay) {
```

Comment:
```suggestion
      if (command === "PRIVMSG" && options.floodDelay > 0) {
```

I generally favor to be as explicit as possible for this kind of test. WDYT?

Reply:
Hum sorry, I did not see that `floodDelay` was undefined by default.

Reply:
I'm thinking, what if the user provides a negative number? That would cause a problem when it's passed to the delay function. Perhaps best to failsafe to no delay in that case.

I've included that check for `> 0`.